frontend/app/src/components/defi/LoanInfo.vue

```diff
@@ -0,0 +1,387 @@
+              <div v-else class="text-right">
+                <premium-lock></premium-lock>
+              </div>
+            </div>
+          </stat-card>
+        </v-col>
+      </v-row>
+      <v-row class="mt-6">
+        <v-col cols="12">
+          <premium-card v-if="!premium" title="Loan History"></premium-card>
+          <vault-events-list
+            v-else
+            :asset="vault.collateralAsset"
+            :events="vault.events"
+            :creation="vault.creationTs"
+            @open-link="openLink($event)"
+          ></vault-events-list>
+        </v-col>
+      </v-row>
+    </v-col>
+    <watcher-dialog
+      :display="showWatcherDialog"
+      title="Collateralization watchers"
+      :message="watcherMessage"
+      :watcher-content-id="watcherVaultId"
+      :existing-watchers="watchers"
+      preselect-watcher-type="makervault_collateralization_ratio"
+      watcher-value-label="Collateralization Ratio"
+      @confirm="addWatcher()"
+      @cancel="showWatcherDialog = false"
+    >
+    </watcher-dialog>
+  </v-row>
+  <v-row v-else align="center" justify="center">
+    <v-col>
+      <span class="font-weight-light subtitle-2">
+        Please select a loan to see information
+      </span>
+    </v-col>
+  </v-row>
+</template>
+
+<script lang="ts">
+import { Component, Prop, Vue } from 'vue-property-decorator';
+import { createNamespacedHelpers } from 'vuex';
+import BaseExternalLink from '@/components/base/BaseExternalLink.vue';
+import WatcherDialog from '@/components/dialogs/WatcherDialog.vue';
+import AmountDisplay from '@/components/display/AmountDisplay.vue';
+import PremiumCard from '@/components/display/PremiumCard.vue';
+import StatCard from '@/components/display/StatCard.vue';
+import HashLink from '@/components/helper/HashLink.vue';
+import PremiumLock from '@/components/helper/PremiumLock.vue';
+import { WatcherArgs } from '@/services/types-api';
+import { MakerDAOVault, Watcher } from '@/services/types-model';
+import { MakerDAOVaultModel } from '@/store/balances/types';
+import { VaultEventsList } from '@/utils/premium';
+
+const { mapState, mapGetters } = createNamespacedHelpers('session');
+const { mapGetters: mapBalanceGetters } = createNamespacedHelpers('balances');
+
+@Component({
+  components: {
+    BaseExternalLink,
+    PremiumCard,
+    PremiumLock,
+    HashLink,
+    AmountDisplay,
+    StatCard,
+    VaultEventsList,
+    WatcherDialog
+  },
+  computed: {
+    ...mapState(['premium']),
+    ...mapGetters(['dateDisplayFormat']),
+    ...mapBalanceGetters(['loanWatchers'])
+  }
+})
+export default class LoanInfo extends Vue {
+  @Prop({ required: true })
+  vault!: MakerDAOVault | MakerDAOVaultModel | null;
+
+  premium!: boolean;
+  dateDisplayFormat!: string;
+  showWatcherDialog: boolean = false;
+  watcherMessage: string = '';
+  watcherVaultId: number | null = null;
+  loanWatchers!: Watcher[];
+
+  get watchers(): Watcher[] {
+    if (this.vault) {
```

Comment:
you could reverse here to remove the indentation from the code below by making it `if (!this.vault) { return []; }` and then move the if clause code directly into the method